plugins/scanners/fossid/src/main/kotlin/FossIdUrlProvider.kt

```diff
@@ -103,8 +103,9 @@ class FossIdUrlProvider private constructor(
             // In order to have a valid URL, the variables must be replaced by some credentials first.
             val replacedUrl = replaced.insertCredentials(UNKNOWN_CREDENTIALS)
 
-            return queryAuthenticator(replacedUrl)?.let { replaced.insertCredentials(it) }
-                ?: replacedUrl.replaceCredentialsInUri()
+            return queryAuthenticator(replacedUrl.replaceCredentialsInUri())?.let {
```

Comment:
Is the `insertCredentials()` call in line 104 now still required? It seems a bit odd to first insert and right away replace credentials again.

Reply:
Done, and I slightly reworded the comment.